hoomd/hpmc/shape_move.py

```diff
@@ -0,0 +1,281 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Shape moves for a for alchemical simulations in extended ensembles."""
+
+import hoomd
+from hoomd.operation import _HOOMDBaseObject
+from . import _hpmc
+from hoomd.hpmc import integrate
+from hoomd.data.parameterdicts import ParameterDict, TypeParameterDict
+from hoomd.data.typeparam import TypeParameter
+import numpy
+
+
+class ShapeMove(_HOOMDBaseObject):
+    """Base class for all shape moves.
+
+    Args:
+        move_probability (`float`): Probability of performing a shape move.
+
+    Attributes:
+        move_probability (`float`): Probability of performing a shape move.
+
+    Note:
+        The document for each subclass details how `move_probability`
+        applies to the move.
+
+    Note:
+        See the documentation of each derived class for a list of supported
+        shapes.
+
+    Note:
+        This class should not be instantiated by users. The class can be used
+        for `isinstance` or `issubclass` checks.
```

Comment:
```suggestion
    Note:
        The document for each subclass details how `move_probability`
        applies to the move.

    Note:
        See the documentation of each derived class for a list of supported
        shapes.

    Note:
        This class should not be instantiated by users. The class can be used
        for `isinstance` or `issubclass` checks.

    Attributes:
        move_probability (`float`): Probability of performing a shape move.
```
I find the docs easier to read with attributes after notes.